workflow-controller/controller-service/src/test/java/io/hyscale/controller/directive/DockerfileUpdateTestHandler.java

```diff
@@ -18,50 +18,36 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.fail;
 
-import java.io.IOException;
-
-import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+import org.springframework.stereotype.Component;
 
 import com.fasterxml.jackson.databind.node.ObjectNode;
 
 import io.hyscale.commons.exception.HyscaleException;
 import io.hyscale.commons.utils.WindowsUtil;
-import io.hyscale.controller.ControllerTestInitializer;
 import io.hyscale.controller.directive.impl.DockerfileJsonHandler;
-import io.hyscale.controller.util.ServiceSpecTestUtil;
 import io.hyscale.servicespec.commons.fields.HyscaleSpecFields;
 import io.hyscale.servicespec.commons.model.service.Dockerfile;
 import io.hyscale.servicespec.commons.model.service.ServiceSpec;
 
-@SpringJUnitConfig(classes = ControllerTestInitializer.class)
-public class DockerfileJsonHandlerTest {
+@Component
+public class DockerfileUpdateTestHandler implements ServiceSpecUpdateTestHandler {
 
     private static final String JSON_PATH = HyscaleSpecFields.getPath(HyscaleSpecFields.image,
             HyscaleSpecFields.dockerfile);
 
+    private static final String dockerfileUpdateSpec = "/servicespecs/dockerfile_update.hspec.yaml";
+
     @Autowired
     private DockerfileJsonHandler dockerfileHandler;
 
-    private static ObjectNode serviceSpecNode = null;
-
-    private static ServiceSpec oldServiceSpec = null;
-
-    private static ServiceSpec updatedServiceSpec = null;
-
-    @Test
-    public void updateDockerfile() {
-        try {
-            oldServiceSpec = ServiceSpecTestUtil.getServiceSpec("/servicespecs/dockerfile_update.hspec.yaml");
-            serviceSpecNode = (ObjectNode) ServiceSpecTestUtil
-                    .getServiceSpecJsonNode("/servicespecs/dockerfile_update.hspec.yaml");
-            dockerfileHandler.update(serviceSpecNode);
-            updatedServiceSpec = new ServiceSpec(serviceSpecNode);
-        } catch (IOException | HyscaleException e) {
-            fail();
-        }
+    @Override
+    public String getServiceSpec() {
+        return dockerfileUpdateSpec;
+    }
 
+    @Override
+    public boolean performValidation(ServiceSpec oldServiceSpec, ServiceSpec updatedServiceSpec) {
```

Comment:
These validation can be on the implementor's entity object rather than on the service spec node. This test is more similar to the actual implementation
Method signature can be of something like this:
 boolean validate(T old, T updated)